diff --git a/astropy/utils/setup_package.py b/astropy/utils/setup_package.py
index 77125ea0b..4067959a5 100644
--- a/astropy/utils/setup_package.py
+++ b/astropy/utils/setup_package.py
@@ -1,7 +1,7 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
 
 from setuptools import Extension
-from os.path import dirname, join, abspath
+from os.path import dirname, join, relpath
 
 ASTROPY_UTILS_ROOT = dirname(__file__)
 
@@ -9,5 +9,5 @@ ASTROPY_UTILS_ROOT = dirname(__file__)
 def get_extensions():
     return [
         Extension('astropy.utils._compiler',
-                  [abspath(join(ASTROPY_UTILS_ROOT, 'src', 'compiler.c'))])
+                  [relpath(join(ASTROPY_UTILS_ROOT, 'src', 'compiler.c'))])
     ]
